src/lib/utils/exception.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
```

Comment:
I stand by my point that this file should be called invalid_input_exception.hpp
Also, the cpp is empty and not needed.